cpp/src/parquet/properties.h

```diff
@@ -190,7 +193,14 @@ class PARQUET_EXPORT ColumnProperties {
 
   size_t max_statistics_size() const { return max_stats_size_; }
 
-  int compression_level() const { return compression_level_; }
+  int compression_level() const { return codec_options_->compression_level_; }
+
+  const std::shared_ptr<CodecOptions>& codec_options() const {
+    if (!codec_options_) {
+      return std::make_shared<CodecOptions>();
```

Comment:
This is a problem:
```
C:\projects\arrow\cpp\src\parquet\properties.h(200) : error C2220: the following warning is treated as an error
C:\projects\arrow\cpp\src\parquet\properties.h(200) : warning C4172: returning address of local variable or temporary
ninja: build stopped: subcommand failed.
```

Reply:
this seems to be fixed, but there are Python errors that I have no ideas on the hints